src/ts/components/ecomodecontainer.ts

```diff
@@ -0,0 +1,126 @@
+import { MediaType, PlayerAPI, SegmentPlaybackEvent, VideoQuality } from 'bitmovin-player';
+import { i18n } from '../localization/i18n';
+import { Container, ContainerConfig } from './container';
+import { EcoModeToggleButton } from './ecomodetogglebutton';
+import { Label, LabelConfig } from './label';
+import { SettingsPanelItem } from './settingspanelitem';
+
+export class EcoModeContainer extends Container<ContainerConfig> {
+  private ecoModeSavedEmissionsItem: SettingsPanelItem;
+  private ecoModeToggleButtonItem: SettingsPanelItem;
+  private emissionsSavedLabel: Label<LabelConfig>;
+  private savedEmissons = 0;
+  private currentEnergyEmission: number;
+
+  constructor(config: ContainerConfig = {}) {
+    super(config);
+
+    const ecoModeToggleButton = new EcoModeToggleButton();
+    const labelEcoMode = new Label({
+      text: i18n.getLocalizer('ecoMode.title'),
+      for: ecoModeToggleButton.getConfig().id,
+      id: 'ecomodelabel',
+    });
+    this.emissionsSavedLabel = new Label({
+      text: this.savedEmissons.toFixed(4) + ' gCO2',
+      cssClass: 'ui-label-savedEnergy',
+    });
+
+    this.ecoModeToggleButtonItem = new SettingsPanelItem(labelEcoMode, ecoModeToggleButton);
+    this.ecoModeSavedEmissionsItem = new SettingsPanelItem('Saved Emissions', this.emissionsSavedLabel, {
+      hidden: true,
+    });
+
+    this.addComponent(this.ecoModeToggleButtonItem);
+    this.addComponent(this.ecoModeSavedEmissionsItem);
+
+    ecoModeToggleButton.onToggleOn.subscribe(() => {
+      this.ecoModeSavedEmissionsItem.show();
+      this.onActivecallback();
+    });
+
+    ecoModeToggleButton.onToggleOff.subscribe(() => {
+      this.ecoModeSavedEmissionsItem.hide();
+      this.onActivecallback();
+    });
+  }
+
+  private onActivecallback: () => void;
+
+  public setOnActiveChangeCallback(callback: () => void) {
+    this.onActivecallback = callback;
+  }

```

Comment:
The naming does not really conclude the meaning. Also, I'm not clear what this is needed for. Maybe we can get around it :)

Reply:
Found out. Callback is used to keep it generic, but in general we want to trigger something that forces the browser to re-calculate the height of the `SettingsPanel` when we remove / add items. That doesn't happen automatically and causes elements to disappear. Just improved the naming a bit:

[Improve naming](https://github.com/bitmovin/bitmovin-player-ui/pull/571/commits/15234d98bd143dc8840940bef189178b4da857c3)